astropy/table/tests/test_groups.py

```diff
@@ -558,10 +558,7 @@ def test_table_filter():
 
     def all_positive(table, key_colnames):
         colnames = [name for name in table.colnames if name not in key_colnames]
-        for colname in colnames:
-            if np.any(table[colname] < 0):
-                return False
-        return True
+        return all(not np.any(table[colname] < 0) for colname in colnames)
```

Comment:
If we're going this route, maybe combine further,
```
return all(np.all(table[colname] >= 0) for colname in table.colnames if colname not in key_colnames)
```
(though really this is just for a test, so perhaps it does not matter)